temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflow.java

```diff
@@ -139,6 +144,35 @@ public void handleSignal(String signalName, Optional<Payloads> input, long event
         "signal " + signalName, () -> workflowProc.handleSignal(signalName, input, eventId));
   }
 
+  @Override
+  public void handleUpdate(
+      String updateName, Optional<Payloads> input, long eventId, UpdateProtocolCallback callbacks) {
+    runner.executeInWorkflowThread(
+        "update " + updateName,
+        () -> {
+          // Skip validator on replay
+          if (!callbacks.isReplaying()) {
+            try {
+              // TODO(https://github.com/temporalio/sdk-java/issues/1748) handleValidateUpdate
+              // should not just be run
+              // in a workflow thread
+              workflowProc.handleValidateUpdate(updateName, input, eventId);
+            } catch (Exception e) {
+              callbacks.reject(this.dataConverter.exceptionToFailure(e));
+              return;
+            }
+          }
+          callbacks.accept();
+          try {
+            Optional<Payloads> result =
+                workflowProc.handleExecuteUpdate(updateName, input, eventId);
+            callbacks.complete(result, null);
+          } catch (Exception e) {
```

Comment:
Hrmm, it is a curious question whether an update exception deserves to fail the update or fail the task. In my head I would think, like workflow functions and signal handlers IIUC, any non-temporal-failure exception is a task failure not an update failure. But for queries it would make sense that any exception is a query failure.

Sure some of us are not fans of the task-failure concept when applied to generic user exceptions, but I think we need to be consistent. So I think any validator exception is a rejection, but exceptions in the update handler itself are treated like exceptions in workflow or signal handlers.

Reply:
Not exactly clear if you're asking for a change here, @cretz but the current implementation to treat exceptions from update execution as failures _of the update_ seems correct to me.

Reply:
It does not seem correct to me and I think @Quinn-With-Two-Ns and I agreed to change this.

In non-Go SDKs, only Temporal exceptions fail the thing (workflow, signal handler, etc) and non-Temporal exceptions fail the task. Task failure vs workflow/update failure is an annoying thing to deal with as a user, but it is consistent with the choices we've made elsewhere.

Reply:
yeah I am not a huge fan o favas decision to make exceptions fail the WFT, but that is the decision so I agree we should be consistent.

Reply:
I misunderstood the original statement but I agree with the the above two comments :+1: